src/lambda-handler.ts

```diff
@@ -0,0 +1,76 @@
+import { APIGatewayProxyEventV2 } from 'aws-lambda';
+import { Args, getS3File, resizeBuffer } from './lib.js';
+import { config } from 'process';
+import { S3ClientConfig } from '@aws-sdk/client-s3';
+
+type ResponseStream = {
+	setContentType(type: string): void;
+	write(stream: string | Buffer): void;
+	end(): void;
+};
+
+type StreamifyHandler = (event: APIGatewayProxyEventV2, response: ResponseStream) => Promise<any>;
+
+const streamify_handler: StreamifyHandler = async (event, response) => {
+	const region = process.env.S3_REGION!;
+	const bucket = process.env.S3_BUCKET!;
+	const config: S3ClientConfig & { bucket: string } = {
+		region: region,
+		bucket: bucket,
+	};
+	if ( process.env.S3_ENDPOINT ) {
+		config.endpoint = process.env.S3_ENDPOINT;
+	}
+
+	if ( process.env.S3_FORCE_PATH_STYLE ) {
+		config.forcePathStyle = true;
+	}
+
+	const key = decodeURIComponent(event.rawPath.substring(1)).replace('/tachyon/', '/');
+	const args = (event.queryStringParameters || {}) as unknown as Args & {
+		'X-Amz-Expires'?: string;
+	};
+	args.key = key;
+	if (typeof args.webp === 'undefined') {
+		args.webp = !!(event.headers && Object.keys(event.headers).find((key) => key.toLowerCase() == 'x-webp'));
+	}
+
+	let s3_response = await getS3File(config, key, args);
+	if (!s3_response.Body) {
+		throw new Error('No body in file.');
+	}
+	let buffer = Buffer.from(await s3_response.Body.transformToByteArray());
+
+	let { info, data } = await resizeBuffer(buffer, args);
+	// If this is a signed URL, we need to calculate the max-age of the image.
+	let maxAge = 31536000;
+	if (args['X-Amz-Expires']) {
+		// Date format of X-Amz-Date is YYYYMMDDTHHMMSSZ, which is not parsable by Date.
+		// @ts-ignore
```

Comment:
Why is this ignored?

Reply:
I think because `args['X-Amz-Date']` is set if `args['X-Amz-Expires']` and I wasn't sure how to tell TS that!

Reply:
Changed to `args['X-Amz-Date']!`